frontend/src/components/Dashboard/MainPanel.js

```diff
@@ -0,0 +1,261 @@
+                snackbarMessage: `Names must match ${NAME_REGEX}, was ${this.state.newWorksheetName}`,
+                snackbarVariant: 'error',
+            });
+            return;
+        }
+
+        executeCommand(`new ${this.state.newWorksheetName || kDefaultWorksheetName}`)
+            .then((data) => {
+                if (data.structured_result && data.structured_result.ui_actions) {
+                    data.structured_result.ui_actions.forEach(([action, param]) => {
+                        if (action === 'openWorksheet') {
+                            window.location.href = `/worksheets/${param}`;
+                        }
+                    });
+                }
+            })
+            .fail((error) => {
+                this.setState({
+                    snackbarShow: true,
+                    snackbarMessage: error.responseText,
+                    snackbarVariant: 'error',
+                });
+            });
+    }
+    /** Renderer. */
+    render() {
+        const { classes } = this.props;
+        return (
+            <div>
+                <Card elevation={0} style={{ height: '100%', backgroundColor: '#f1f1f1' }}>
+                    <Box className={classes.titleBox} display={'flex'} alignItems={'center'}>
+                        <h3 className={classes.heading}>Worksheets</h3>
+                        &nbsp;&nbsp;&nbsp;&nbsp;
+                        <Tooltip title='New Worksheet'>
+                            <Button
+                                variant='contained'
+                                className={classes.button}
+                                startIcon={<NewWorksheetIcon />}
+                                onClick={() => this.setState({ newWorksheetShowDialog: true })}
+                            >
+                                ADD
+                            </Button>
+                        </Tooltip>
+                    </Box>
+
+                    {this.state.worksheets}
+                </Card>
+                <Dialog
+                    open={this.state.newWorksheetShowDialog}
+                    onClose={() => this.resetDialog()}
+                    aria-labelledby='form-dialog-title'
+                    style={{ backgroundColor: '#f1f1f1' }}
+                >
+                    <DialogTitle id='form-dialog-title'>New Worksheet</DialogTitle>
+                    <DialogContent>
+                        <TextField
+                            autoFocus
+                            margin='dense'
+                            id='name'
+                            label='Name'
+                            fullWidth
+                            value={this.state.newWorksheetName}
+                            placeholder={this.state.newWorksheetName}
+                            onChange={(e) => this.setState({ newWorksheetName: e.target.value })}
+                            onKeyDown={(e) => {
+                                if (e.keyCode === 13) {
```

Comment:
Maybe we declare somewhere a enum mapping like ENTER_KEY_CODE = 13, ESC_KEY_CODE = 27 so we can reuse later, but can do it later